pkg/middleware/interceptor.go

```diff
@@ -103,8 +103,9 @@ func AsGRPCError(err error) error {
 		errors.Is(err, handler.ErrResourceID),
 		errors.Is(err, handler.ErrSematicVersion),
 		errors.Is(err, handler.ErrConnectorNamespace),
-		errors.Is(err, handler.ErrUpdateMask):
-
+		errors.Is(err, handler.ErrUpdateMask),
+		errors.Is(err, handler.ErrInvalidKnowledgeBaseName),
+		errors.Is(err, customerror.ErrInvalidArgument):
```

Comment:
Hi @Yougigun I saw this PR once merged but I would have liked to add the following comment. It is not a blocker, just for awareness.

I can't find the conversation (I think it happened on a PR like 1 month ago) but I proposed to @donch1989 using a single package to communicate errors through packages. These errors would be a bit more generic (not found, resource exists, invalid argument...) and the package generating the error would be responsible of [wrapping a given error into the domain one](https://github.com/instill-ai/pipeline-backend/blob/main/pkg/service/secret.go#L29). This has several advantages:
- In the interceptor we'd reduce a lot the amount of errors and packages we manage. Right now when we add a feature that returns a given error, we need to remember to handle it in the interceptor.
- We can implement a retry policy handler for Temporal workflows or other type of event handler without copying all the errors that middleware already handles.
- Ideally we could define these errors in `service`, which acts as the domain layer. Sadly, the current dependency injections would cause dependency cycles. However, using a package to communicate errors will sanitize some of this dependencies (in the cases where the package calls a dependency through an interface but it still needs to import the implementation package to handle the error).

Reply:
I like the idea of using a single error package. The reason I have a custom error is also to try and solve the dependency cycle.

Reply:
> The reason I have a custom error is also to try and solve the dependency cycle

Yes I'm feeling that pain, too. One problem we have is that, even if we communicate through interfaces, many times we define them in the implem package instead of in the caller / in a separate package, which produces the same dependencies as if we were using a struct directly.